src/devices/Card/Mifare/MifareCard.cs

```diff
@@ -609,5 +673,673 @@ private byte[] Serialize()
+                KeyB = newKeyB.ToArray();
+                newKeyB.CopyTo(Data.AsSpan().Slice(10, 6));
+            }
+
+            // Find the sector tailer based on the sector number and it's a safe byte cast
+            var success = WriteDataBlock(sectorTailer);
+            if (!success)
+            {
+                return false;
+            }
+
+            byte firstBlock = (byte)(sector >= 32 ? 32 * 4 + (sector - 32) * 16 : sector * 4);
+            // Authenticate to the rest of the blocks to format and format them
+            for (byte block = firstBlock == 0 ? firstBlock++ : firstBlock; block < (firstBlock + nbBlocks - 1); block++)
+            {
+                BlockNumber = block;
+                Command = authenticateWithKeyA ? MifareCardCommand.AuthenticationA : MifareCardCommand.AuthenticationB;
+                ret = RunMifareCardCommand();
+                if (ret < 0)
+                {
+                    return false;
+                }
+
+                Data = new byte[16];
+                success = WriteDataBlock(block);
+                if (!success)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Select the card. Needed if authentication or read/write failed
+        /// </summary>
+        /// <returns>True if success</returns>
+        public bool ReselectCard()
+        {
+            return _rfid.ReselectTarget(Target);
+        }
+
+        /// <summary>
+        /// Format the Card to NDEF
+        /// </summary>
+        /// <returns>True if success</returns>
+        public bool FormatNdef()
+        {
+            return FormatNdef(new byte[0]);
+        }
+
+        /// <summary>
+        /// Format the Card to NDEF
+        /// </summary>
+        /// <param name="keyB">The key B to be used for formatting, if empty, will use the default key B</param>
+        /// <returns>True if success</returns>
+        public bool FormatNdef(ReadOnlySpan<byte> keyB)
+        {
+            if (Capacity is not MifareCardCapacity.Mifare1K or MifareCardCapacity.Mifare2K or MifareCardCapacity.Mifare4K)
+            {
+                throw new ArgumentException($"Only Mifare card classic are supported with capacity of 1K, 2K and 4K");
+            }
+
+            int nbBlocks = GetNumberBlocks();
+
+            byte[] keyFormat = keyB.Length == 0 ? DefaultKeyB : keyB.ToArray();
+            if (keyFormat.Length != 6)
+            {
+                throw new ArgumentException($"{nameof(keyB)} can only be empty or 6 bytes length");
+            }
+
+            // First write the data for the format
+            // All block data coming from https://www.nxp.com/docs/en/application-note/AN1304.pdf page 30+
+            var authOk = AuthenticateBlockKeyB(keyFormat, 1);
+            Data = new byte[] { 0x14, 0x01, 0x03, 0xE1, 0x03, 0xE1, 0x03, 0xE1, 0x03, 0xE1, 0x03, 0xE1, 0x03, 0xE1, 0x03, 0xE1 };
```

Comment:
should the array be static readonly somewhere here? this way you'll avoid lots of allocations, also you'd give them names and it would be more clearer what's actually happening here

Reply:
Yes, I can definitely do that. And I'm using it 2 times. So definitely makes sense.